File: focus_todo_mobile/lib/sync/outbound_selector.dart
import '../shared/models/task.dart';
import '../shared/models/task_sync_metadata.dart';

List<Task> getOutboundTasks(
  List<Task> tasks,
  List<TaskSyncMetadata> syncMetadata,
) {
  final pendingTaskIds = syncMetadata
      .where(
        (TaskSyncMetadata item) =>
            item.syncStatus == TaskSyncStatus.pendingUpsert ||
            item.syncStatus == TaskSyncStatus.pendingDelete,
      )
      .map((TaskSyncMetadata item) => item.taskId)
      .toSet();

  return tasks
      .where((Task task) => pendingTaskIds.contains(task.id))
      .toList(growable: false);
}

File: focus_todo_computer/focus-todo/src/sync/outboundSelector.ts
import type { Task, TaskSyncMetadata } from '../types/task'

export function getOutboundTasks(
  tasks: Task[],
  syncMetadata: TaskSyncMetadata[],
): Task[] {
  const pendingTaskIds = new Set(
    syncMetadata
      .filter(
        (item) =>
          item.syncStatus === 'pending-upsert' ||
          item.syncStatus === 'pending-delete',
      )
      .map((item) => item.taskId),
  )

  return tasks.filter((task) => pendingTaskIds.has(task.id))
}
